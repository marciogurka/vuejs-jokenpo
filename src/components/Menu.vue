<template>
  <div>
    <router-link to="game" exact
      class="button is-success is-outlined is-large is-fullwidth"
      id="start-game">
      <span class="icon is-large">
        <i class="fas fa-hand-scissors"></i>
      </span>
      <span>Start a game</span>
    </router-link>
    <a class="button is-outlined is-large is-fullwidth" href="https://pt.wikipedia.org/wiki/Pedra-papel-tesoura-lagarto-Spock" target="_blank">
      <span class="icon is-large">
        <i class="fas fa-question-circle"></i>
      </span>
      <span>What's Jokenpo?</span>
    </a>
    <a class="button is-outlined is-large is-fullwidth" href="https://github.com/marciogurka/vuejs-jokenpo" target="_blank">
      <span class="icon is-large">
        <i class="fas fa-code"></i>
      </span>
      <span>See the source code</span>
    </a>
  </div>
</template>

<script>
export default {
  name: 'Menu',
};
</script>

<!-- Add "scoped" attribute to limit CSS to this component only -->
<style scoped>
.button {
  margin-bottom: 30px;
}

.button:last-child {
  margin-bottom: 0;
}

@media screen and (max-width: 425px) {
  div > .button {
    font-size: large;
  }
}


</style>
