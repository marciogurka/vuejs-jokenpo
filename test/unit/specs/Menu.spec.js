import { mount, RouterLinkStub } from '@vue/test-utils';
import Menu from '@/components/Menu';

describe('MenuSpec.vue', () => {
  let cmp;

  /* Mouting the component before each test */
  beforeEach(() => {
    cmp = mount(Menu, {
      stubs: {
        RouterLink: RouterLinkStub,
      },
    });
  });

  it('has the expected html structure', () => {
    expect(cmp.element).toMatchSnapshot();
  });

  it('has a /game link option', () => {
    expect(cmp.find('.button').attributes().id).toBe('start-game');
  });
});
