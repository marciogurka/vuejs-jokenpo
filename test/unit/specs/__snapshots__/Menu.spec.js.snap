// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MenuSpec.vue has the expected html structure 1`] = `
<div>
  <a
    class="button is-success is-outlined is-large is-fullwidth"
    id="start-game"
  >
    <span
      class="icon is-large"
    >
      <i
        class="fas fa-hand-scissors"
      />
    </span>
     
    <span>
      Start a game
    </span>
  </a>
   
  <a
    class="button is-outlined is-large is-fullwidth"
    href="https://pt.wikipedia.org/wiki/Pedra-papel-tesoura-lagarto-Spock"
    target="_blank"
  >
    <span
      class="icon is-large"
    >
      <i
        class="fas fa-question-circle"
      />
    </span>
     
    <span>
      What's Jokenpo?
    </span>
  </a>
   
  <a
    class="button is-outlined is-large is-fullwidth"
    href="https://github.com/marciogurka/vuejs-jokenpo"
    target="_blank"
  >
    <span
      class="icon is-large"
    >
      <i
        class="fas fa-code"
      />
    </span>
     
    <span>
      See the source code
    </span>
  </a>
</div>
`;
